proxify/proxify

```diff
@@ -1,81 +1,128 @@
 
-	src="${1}"
-	len="${#src}"
+	src="${1}"  # get the first argument of this function
+	len="${#src}"  # get the length of src
 
 	for (( pos = 0; pos < len; pos++ ))
 	do
-		ch="${src:${pos}:1}"
+		ch="${src:${pos}:1}"  # ch = get 1 character from pos-th position of src
+
+		# case <variable> in
+		# pattern)
+		# 	<commands>
+		# ;;
 
-		case "${ch}" in
-			[-_.~a-zA-Z0-9])
-				cv="${ch}"
+		case "${ch}" in  
+			[-_.~a-zA-Z0-9])  # if ch is one of the following characters
+				cv="${ch}"  # let it be ~~~
 				;;
 
-			*)
-				printf -v cv '%%%02x' "'${ch}"
+			*)  # else
+				printf -v cv '%%%02x' "'${ch}"  # url encode ch using specified format, and put it in cv
+
+				# %% prints %
+				# %02x prints 2 digits of the hexadecimal representation of ch
 				;;
 		esac
 
-		buf="${buf}${cv}"
+		buf="${buf}${cv}"  # append encoded character to buffer
 	done
 
-	echo "${buf}"
+	echo "${buf}"  # return result
 }
 
-function ask_host ()
+function ask_host()
 {
-	local host port
+	local host port  # local variables
 
-	read -p "Host: " host
-	read -p "Port: " port
+	read -p "Host: " host  # prompt for host
+	read -p "Port: " port  # prompt for port
 
-	echo "$host:$port"
+	echo "$host:$port"  # return result
 }
 
-function ask_auth ()
+function ask_pass()
 {
-	local user pass
+	local pass  # local variables
 
-	read -p "User: " user
-	read -p "Pass: " -s pass
-	echo > /dev/stderr
+	read -p "Pass: " -s pass  # promp for password, hide characters
+	echo > /dev/stderr    # prevent newline from being returned
 
-	echo "$(url_encode ${user}):$(url_encode ${pass})"
+	echo "$(url_encode ${pass})"  # return the value of url-encoded password
 }
 
-function gen_base ()
+function ask_auth()
 {
-	local host auth is_a base
+	local user pass  # local variables
+
+	read -p "User: " user  # prompt for username
+	pass="$(ask_pass)"  # ask for password
+
+	echo "$(url_encode ${user}):${pass}"  # return the value of url-encoded username appended by password
+}
 
-	host="$(ask_host)"
+function gen_base()
+{
+	local host auth is_a base ask_pass_only  # local variables
 
-	read -n1 -p "Authentificate? [Y/N] " is_a
-	echo > /dev/stderr
+	ask_pass_only="$1"
 
-	if [ "${is_a^^}" == "Y" ]
+	if [[ "$ask_pass_only" == true ]]
 	then
-		auth="$(ask_auth)"
-		base="$auth@$host"
+		base="$(ask_pass)"  # ask for password only
 	else
-		base="$host"
+		host="$(ask_host)"  # ask host
+
+		read -n1 -p "Authenticate? [Y/N] " is_a  # want to use auth? (read one character only)
+		echo > /dev/stderr  # prevent newline from being returned
+
+		if [ "${is_a^^}" == "Y" ]  # if use auth (convert the answer to uppercase)
+		then
+			auth="$(ask_auth)"  # ask for username and password
+			base="$auth@$host"  # prepend credentials to base
+		else
+			base="$host"  # no auth, no credentials needed
+		fi
 	fi
 
-	echo "$base"
+	echo "$base"  # return result
 }
 
 # Main
 
-base="$(gen_base)"
+conf_file=$(echo ~/proxify.conf)  # path of configuration file
```

Comment:
Please rename this to `.proxify.conf`, so users don't get stray config. files around.